File: src/028/p028.erl
-module(p028).
-export([solve/0]).

spiral_diagonal_sum(1) -> 1;
spiral_diagonal_sum(X) when X > 0 -> 4*X*X - 6*(X-1) + spiral_diagonal_sum(X-2).

solve() -> spiral_diagonal_sum(1001).

File: src/028/p028.exs
defmodule Problem028 do
  defp spiral_diag_sum(1), do: 1
  defp spiral_diag_sum(2), do: 10
  defp spiral_diag_sum(x) when is_integer(x) and x > 0, do: 4*x*x - 6*(x-1) + spiral_diag_sum(x-2)

  def solve, do: spiral_diag_sum(1001)
end

IO.puts Problem028.solve
